AsyncQueue.test.ts

```diff
@@ -0,0 +1,87 @@
+import { AsyncQueue } from "./AsyncQueue";
+
+const uploadFile = (fileName: string, file: string) =>
+  new Promise<void>((resolve) => {
+    logWrapper(`Started uploading: ${fileName}, ${file}`);
+    setTimeout(() => {
+      logWrapper(`Done uploading: ${fileName}`);
+      resolve();
+    }, 1000);
+  });
+
+async function cleanUp(queue: AsyncQueue, timeout = 3000) {
+  queue.stop();
+  await new Promise((resolve) => setTimeout(resolve, timeout));
+}
```

Comment:
can you explain what are you trying to do here?
if this function is to w8 that the queue will finish please find a better way to do that.
maybe understand if all the tasks have been executed.

Reply:
Done